### DISCUSSION TITLE
Add viewer/open app to a tkinter app like extented

### DISCUSSION THREAD

**Discussion by MaxNiark** [TomSchimansky/CustomTkinter#2216] [General]
Hii

i'm looking to perform my app with other support like viewer. 

obvisouly tkinter and this custom is not making to see video or pdf (do search, skip moment in a video, zoom in picture and keep the pixel ;) ) so i need some library in python, or any other open app to link with my app. 

PDF viewer 
Video viewer
Picture viewer

thank for your help :)

**Comment by Akascape** [comment]
@MaxNiark We already have these things
**PDF Viewer**: https://github.com/Akascape/CTkPDFViewer

![example](https://github.com/Akascape/CTkPDFViewer/assets/89206401/1324243e-da47-4bd8-af46-cded53cb7b51)

**VIdeo Player**: https://github.com/Akascape/tkVideoPlayer

https://user-images.githubusercontent.com/89206401/228515652-abe137d0-6823-4c56-ba5c-8eb8292e0182.mp4

**Image Viewer example**: https://github.com/TomSchimansky/CustomTkinter/discussions/2212#discussioncomment-8171509
Image viewer can easily be made to zoom using the mousewheel bind with CTkImage size configure.

**Comment by MaxNiark** [comment]
hi Akascape 

unfortunately i don't like the pdf tkinter viewer cuz i can't process the find text and other thing (ctrl+f) 

EDIT : Also the pdf tkinter viewer is soo gluttonous with my pdf and the memory 

i'll check for the video, that sound good. 

for the picture viewer u cannot open a list of img. it necessary to save the list of path with panda lib or else and process a code with pass/next button and other.  hard to work, someone have maked something good, i'm sure ! ^^ 

Play an .exe from my appp, with the data_path as argument is the good way, but windog software is lock as apple system brruh

**Reply by Akascape** [reply]
I guess you will have to make those yourself.

The video player should work without much memory consumption.
